lib/ansible/module_utils/urls.py

```diff
@@ -716,10 +720,15 @@ def open_url(url, data=None, headers=None, method=None, use_proxy=True,
             handlers.append(authhandler)
 
         elif username and force_basic_auth:
-            if headers is None:
-                headers = {}
+            headers["Authorization"] = basic_auth_header(username, password)
 
-            headers["Authorization"] = "Basic %s" % base64.b64encode("%s:%s" % (username, password))
+        else:
+            rc = netrc.netrc(os.environ.get('NETRC'))
+            login = rc.authenticators(parsed[1])
```

Comment:
this assumes NETRC is always present, you should make it conditional on it actually resulting in something